### PR TITLE
Collapse long URLs in rendered notes

### LINKED ISSUE #93: Handle super-long URLs
Find a way to make long URLs (multiple lines of text on max page width) to look nice.

If there is an existing well-maintained lib for shortening URLs with "...", consider using it over implementing.

### PR DESCRIPTION
**Pull request by dreikanter**
## Summary

Add intelligent URL shortening to note rendering that displays long URLs with a middle ellipsis while preserving the full URL in the href and title attributes. This applies to both GFM autolinks (`<url>` syntax) and explicit markdown links where the visible text matches the destination URL.

### Changes

- **Autolink rendering**: Added `renderAutoLink()` to handle `ast.KindAutoLink` nodes with automatic shortening of URLs exceeding 60 characters
- **Explicit link handling**: Extended `renderLink()` to detect when a link's visible text equals its destination and apply the same shortening logic
- **URL shortening algorithm**: Implemented `shortenURL()` that preserves the first 2/3 of the URL (scheme, host, leading path) and the final 1/3 (filename/query), joining them with a middle ellipsis
- **CSS fallback**: Added `overflow-wrap: anywhere` to `.markdown-body a` to handle edge cases where long URLs aren't shortened
- **Documentation**: Updated CHANGELOG.md to reflect the feature

### Testing

Added comprehensive test coverage:
- `TestLongAutoLinkShortened`: Verifies autolinks are shortened with title attribute
- `TestShortAutoLinkUnchanged`: Confirms short URLs render unchanged
- `TestLongExplicitLinkWithMatchingTextShortened`: Tests `[url](url)` syntax shortening
- `TestExplicitLinkWithCustomTextUntouched`: Ensures custom link text is never modified
- `TestShortenURL`: Unit tests for the shortening algorithm with edge cases (boundary conditions, unicode handling)

## References

- Closes https://github.com/dreikanter/notes-view/issues/93

### CODE REVIEW

File: internal/renderer/noteext.go [hunk 1]
```diff
@@ -204,47 +204,137 @@ type noteLinkRenderer struct{}
...
+	if shortened != display {
+		_, _ = w.WriteString(` title="`)
+		_, _ = w.Write(util.EscapeHTML([]byte(display)))
+		_ = w.WriteByte('"')
+	}
+	_ = w.WriteByte('>')
+	_, _ = w.Write(util.EscapeHTML([]byte(shortened)))
+	_, _ = w.WriteString("</a>")
+	return ast.WalkContinue, nil
 }
```
[the remark this answers is not in the document]
Comment (dreikanter) [reply internal/renderer/noteext.go:262]: Good catch — fixed in 38734e7. Routed `renderAutoLink` through `goldmarkhtml.IsDangerousURL` and extended `TestDangerousURLsSanitized` with autolink-form inputs (`<javascript:…>`, `<vbscript:…>`, `<data:text/html,…>`). Verified the guard is load-bearing by stashing it and observing the test fail.

---
_Generated by [Claude Code](https://claude.ai/code)_
